### DISCUSSION TITLE
Tor won't connect

### DISCUSSION THREAD

**Discussion by GigaVik** [hassio-addons/app-tor#131] [Q&A]
Is there a plugin update planned?
In particular, adding bridge connection settings?
At the moment I can't connect to TOR
`
[cont-init.d] 00-banner.sh: exited 0.
[cont-init.d] 01-log-level.sh: executing... 
[cont-init.d] 01-log-level.sh: exited 0.
[cont-init.d] 10-requirements.sh: executing... 
[cont-init.d] 10-requirements.sh: exited 0.
[cont-init.d] 20-dirs.sh: executing... 
[cont-init.d] 20-dirs.sh: exited 0.
[cont-init.d] 50-tor-log.sh: executing... 
[cont-init.d] 50-tor-log.sh: exited 0.
[cont-init.d] 51-tor-socks.sh: executing... 
[cont-init.d] 51-tor-socks.sh: exited 0.
[cont-init.d] 52-tor-services.sh: executing... 
[cont-init.d] 52-tor-services.sh: exited 0.
[cont-init.d] 90-hostname.sh: executing... 
[11:26:37] INFO: Starting Tor temporarly...`

**Comment by akrigator** [comment]
@GigaVik could you try with bridges https://github.com/hassio-addons/addon-tor/pull/170 ?

**Reply by GigaVik** [reply]
Great! Can you help me? how to install support for this bridge in my configuration? For some reason I couldn't figure it out...

**Reply by akrigator** [reply]
Not sure how to do this throw addons repo, i have used dev approach:
```
cd /addons/
git clone https://github.com/akrigator/addon-tor.git
cd addons/addon-tor/
git checkout origin/obfs
```
After restart home assistant local-tor addon should be available in your, so it can be installed/run/configured as the community version
Only one difference in update - it would be performed via git

**Reply by akrigator** [reply]
I've check obfs  and webtunnel bridges, since it much easy to configuration for them (see addon's configuration after installation). Obfs does not work in russia for now, but webtunnel still work, but it takes about 40 minutes to negotiate connection in my case. 
Snowflake does not work for me as well

**Reply by GigaVik** [reply]
I finally installed your version of the addon, and it works! Although the connection does take a very long time... As you said, about 40 minutes.

**Comment by Alexei-vk2** [comment]
hello!
I installed Tor, but it doesn't start. 
How can I fix this?
![Снимок экрана 2024-04-07 230640](https://github.com/hassio-addons/addon-tor/assets/166062625/fc2d44ab-6759-4b16-aa34-cd1a17f3e523)
![Снимок экрана 2024-04-07 230629](https://github.com/hassio-addons/addon-tor/assets/166062625/870f4b6e-e415-46e3-b560-991498ed3861)

**Reply by GigaVik** [reply]
https://github.com/hassio-addons/addon-tor/discussions/131#discussioncomment-7424792
Have you tried installing the bridge enabled version of the addon?

**Comment by Alexei-vk2** [comment]
Hello!
Faced with this situation:
Casambi integration is installed in the home assistant.
when the "unavailable" condition occurs, the lights turn on and off.

is there any way to make sure that even after the "not available" state, the devices have the last mode so that they are turned off?
can you tell me how to do this?
![WhatsApp Image 2024-05-13 at 10 57 27](https://github.com/hassio-addons/addon-tor/assets/166062625/2f34beb9-89dd-4345-a28b-bc4191e75706)

**Comment by akrigator** [comment]
@GigaVik, i've created a custom repository with forked version of tor addon from hassio-community  repo and added bridge support. Installation is easy as for community
1. Add report https://github.com/hassio-addons/repository to HA
2. Install plugin from new repository
